dot/state/epoch.go

```diff
@@ -301,7 +380,17 @@ func (s *EpochState) GetLatestConfigData() (*types.ConfigData, error) {
 
 // HasConfigData returns whether config data exists for a given epoch
 func (s *EpochState) HasConfigData(epoch uint64) (bool, error) {
-	return s.db.Has(configDataKey(epoch))
+	has, err := s.db.Has(configDataKey(epoch))
+
+	if !has || errors.Is(chaindb.ErrKeyNotFound, err) {
+		s.nextEpochLock.Lock()
+		defer s.nextEpochLock.Unlock()
+
+		_, has = s.nextConfigData[epoch]
+		return has, nil
+	}
+
+	return has, err
```

Comment:
- Less indentation
- Wrap database error if it's not key not found
- I think `has` can never be false from the database

```suggestion
	_, err := s.db.Has(configDataKey(epoch))
	if err == nil {
		return true, nil
	}

	if !errors.Is(chaindb.ErrKeyNotFound, err) {
		return false, fmt.Errorf("cannot check database for config data key of epoch %d: %w", epoch, err)
	}
	
	// Key not found in database
	s.nextEpochLock.Lock()
	defer s.nextEpochLock.Unlock()
	_, has := s.nextConfigData[epoch]
	return has, nil
```